packages/cli/test/unit/commands/secrets.test.ts

```diff
@@ -0,0 +1,37 @@
+import { client } from '../../mocks/client';
+import secrets from '../../../src/commands/secrets';
+import { useSecrets } from '../../mocks/secrets';
+import { useUser } from '../../mocks/user';
+import { useTeams } from '../../mocks/team';
+import ms from 'ms';
+
+describe('secrets', () => {
+  it('errors when no subcommand is provided', async () => {
+    client.setArgv('secrets');
+    const exitCode = await secrets(client);
+
+    expect(exitCode).toEqual(2);
+  });
+
+  it('lists secrets with ls subcommand', async () => {
+    useUser();
+    useTeams('team_dummy');
+    const name = 'secret-api-password';
+    const created = 1519555701;
+    useSecrets({ name, created });
+
+    client.setArgv('secrets', 'ls');
+    await secrets(client);
+
+    const timeAgo = `${ms(
+      new Date().getTime() - new Date(created).getTime()
+    )} ago`;
+    await expect(client.stderr).toOutput(
+      '> NOTE: The `vercel env ls` command is recommended instead of `vercel secret ls`'
+    );
+
+    const output = client.stderr.read();
+    await expect(output).toMatch(name);
+    await expect(output).toMatch(timeAgo);
```

Comment:
This _may_ flake occasionally if you hit a `1000ms` boundary by chance. Sadly

```
jest.useFakeTimers();
jest.setSystemTime(new Date('2020-02-19'));
```

causes the test to hang and timeout due to other timer use in the code. A bit more refactoring will be required to mock time it seems.